selftests/unit/test_datadir.py

```diff
@@ -34,6 +34,54 @@ def _get_temp_dirs_mapping_and_config(self):
         config_file.close()
         return (base_dir, mapping, config_file.name)
 
+    @staticmethod
+    def _make_dummy_registrations(stg):
```

Comment:
I don't think this is the best course of action.  Consider using the *actual* `register_core_options()`, with a different settings instance:

```diff
diff --git a/selftests/unit/test_datadir.py b/selftests/unit/test_datadir.py
index f2efa043..ed33fed9 100644
--- a/selftests/unit/test_datadir.py
+++ b/selftests/unit/test_datadir.py
@@ -34,54 +34,6 @@ class Base(unittest.TestCase):
         config_file.close()
         return (base_dir, mapping, config_file.name)
 
-    @staticmethod
-    def _make_dummy_registrations(stg):
-        """
-        Register dummy options to be updated by the new Settings API
-
-        The new Settings API never register options from config files. This
-        method creates dummy registrations to be updated later with the
-        correct values for the options.
-        """
-        help_msg = 'Cache directories to be used by the avocado test'
-        stg.register_option(section='datadir.paths',
-                            key='cache_dirs',
-                            key_type=list,
-                            default=[],
-                            help_msg=help_msg)
-
-        help_msg = 'Base directory for Avocado tests and auxiliary data'
-        stg.register_option(section='datadir.paths',
-                            key='base_dir',
-                            key_type=os.path.expanduser,
-                            default=os.path.expanduser(
-                                '/var/lib/avocado'),
-                            help_msg=help_msg)
-
-        help_msg = 'Test directory for Avocado tests'
-        stg.register_option(section='datadir.paths',
-                            key='test_dir',
-                            key_type=os.path.expanduser,
-                            default=os.path.expanduser(
-                                '/usr/share/doc/avocado/tests'),
-                            help_msg=help_msg)
-
-        help_msg = 'Data directory for Avocado'
-        stg.register_option(section='datadir.paths',
-                            key='data_dir',
-                            key_type=os.path.expanduser,
-                            default=os.path.expanduser(
-                                '/var/lib/avocado/data'),
-                            help_msg=help_msg)
-
-        help_msg = 'Logs directory for Avocado'
-        stg.register_option(section='datadir.paths',
-                            key='logs_dir',
-                            key_type=os.path.expanduser,
-                            default=os.path.expanduser(
-                                '~/avocado/job-results'),
-                            help_msg=help_msg)
-
     def setUp(self):
         (self.base_dir,
          self.mapping,
@@ -99,7 +51,9 @@ class DataDirTest(Base):
         When avocado.conf is present, honor the values coming from it.
         """
         stg = future_settings.Settings()
-        self._make_dummy_registrations(stg)
+        with unittest.mock.patch('avocado.core.future_settings', stg):
+            import avocado.core
+            avocado.core.register_core_options()
         stg.process_config_path(self.config_file_path)
         stg.merge_with_configs()
         with unittest.mock.patch('avocado.core.data_dir.future_settings', stg):
```

Reply:
I spent a good time thinking on how to register the default values. This is much simpler, thanks! I'll test it next Monday.